staging/src/k8s.io/apiserver/pkg/server/options/audit.go

```diff
@@ -340,10 +346,26 @@ func (o *AuditBatchOptions) AddFlags(pluginName string, fs *pflag.FlagSet) {
 			"moment if ThrottleQPS was not utilized before. Only used in batch mode.")
 }
 
+type safeBackend struct {
+	audit.Backend
+}
+
+func (s *safeBackend) ProcessEvents(ev ...*auditinternal.Event) bool {
+	s.Backend.ProcessEvents(ev...)
+	return true
+}
+
+func (s *safeBackend) String() string {
+	return fmt.Sprintf("safe<%s>", s.Backend)
+}
+
 func (o *AuditBatchOptions) wrapBackend(delegate audit.Backend) audit.Backend {
-	if o.Mode == ModeBlocking {
+	if o.Mode == ModeBlockingStrict {
 		return delegate
 	}
+	if o.Mode == ModeBlocking {
+		return &safeBackend{Backend: delegate}
```

Comment:
I would prefer to have this split explicitly in the handler https://github.com/kubernetes/kubernetes/pull/65763/files#diff-5409fdc4797413d3864890678136abfbR59 instead of doing plugin wrapping here.

Reply:
Makes the code clearer IMO.

Reply:
Actually I find `safe` as a term confusing. Would expect something like `ignoreErrors`.

Reply:
Changed to `ignoreErrors`, I agree `safe` wasn't the best word here.

By explicit split you mean doing check every time an audit event comes? I thought the general approach in the apiserver is to set up everything up front, based on configuration provided.